### DISCUSSION TITLE
Explainer hooks and the need for extra parameters for models that require more than model, x, edge_index and y

### DISCUSSION THREAD

**Discussion by anna-inf-met** [pyg-team/pytorch_geometric#9155] [Q&A]
I would really like to be able to experiment with the explainer tool for regressionXAI but have found that my models all have additional parameters (for example edge_weights) that the explainer global hook (understandably) takes issue with.  Naively, it would seem that this may be easily navigated simply by passing kwargs, but this does not work.  Would anyone be able to offer some advice, or am I simply going to have to do a forward pass re-write for a new bespoke explainer.

**Comment by rusty1s** [comment]
Do you mind sharing a small example to replicate your issue? PyG's explainer can take more arguments than `x`, `edge_index` and `y`, and you should be able to simply pass them as keyword arguments to the explainer:
```python
explainer(x, edge_index, edge_attr=edge_attr, batch=batch)
